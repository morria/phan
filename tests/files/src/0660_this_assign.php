<?php

class Foo660 {
    /** @var ?int description */
    public $prop;

    /**
     * @var ?stdClass
     */
    public $possibly_null;

    // Warning messages include Phan's inferred types
    public function example() {
        $this->prop = 2;
        echo strlen($this->prop);
    }

    public function example2() {
        if ($this->possibly_null) {
            echo strlen($this->possibly_null);
        } else {
            echo "this is null\n";
            echo strlen($this->possibly_null);
        }
    }
}
