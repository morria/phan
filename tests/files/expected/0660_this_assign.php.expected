%s:15 PhanTypeMismatchArgumentInternal Argument 1 (string) is 2 but \strlen() takes string
%s:20 PhanTypeMismatchArgumentInternal Argument 1 (string) is \stdClass but \strlen() takes string
%s:23 PhanTypeMismatchArgumentInternal Argument 1 (string) is null but \strlen() takes string
